### ISSUE TITLE
Clarification on train/val/test split assignment in `stage2_finetune/main_finetune.py`

### ISSUE THREAD

**Issue by Dandy5721** [hzlab/Brain-Harmony#5]
Hi, thanks for releasing the code.

While reading through `modules/harmonizer/stage2_finetune/main_finetune.py`, I noticed that the dataset construction assigns:

```python
dataset_test = ... split="val" ...
dataset_val  = ... split="test" ...
```

i.e. the object called `dataset_test` is built from the `"val"` split, and the object called `dataset_val` is built from the `"test"` split. Before drawing conclusions about the reported numbers I'd like to understand the intended semantics here.

A few related questions:

1. Were Tables 1 and 2 in the paper produced by this script?
2. Which split was used for checkpoint selection in the reported fine-tuning experiments, the object named `dataset_val` (which currently reads from `split="test"`), or the actual validation split?
3. For the downstream normalization parameters: are the statistics fit only on training subjects? I noticed `datasets.py` writes split-specific normalization files, which made me want to double-check.
4. Could you share, in whatever form is compatible with your data use agreements, evidence that the train / val / test splits used for the reported runs are subject-disjoint and reproducible, for example: the split-generation script and seed, a splits.json (or hashes of the per-split subject lists), per-split subject counts, and the source split used for fitting normalization parameters? 

If the variable naming is just an inversion that doesn't affect the experimental flow, a brief confirmation would be very helpful. Otherwise, would it be possible to rerun Tables 1 and 2 with (a) train-only normalization, (b) validation-only checkpoint selection, and (c) a single, untouched test-set evaluation, so the reported numbers can be confirmed?

Thanks again for the work and for making the code available.
